src/hotspot/share/opto/type.hpp

```diff
@@ -2166,6 +2169,15 @@ inline bool Type::is_floatingpoint() const {
   return false;
 }
 
+template <>
+inline const TypeInt* Type::is<TypeInt>() const {
```

Comment:
I think `cast` would be a better name

Reply:
I named it `is` for consistency with all the other `is_*` methods but, of course, `cast` would be much less confusing. I'm not sure if we should be consistent or give this a more understandable name.